app_dart/lib/src/request_handlers/update_agent_health.dart

```diff
@@ -45,22 +49,55 @@ class UpdateAgentHealth extends ApiRequestHandler<UpdateAgentHealthResponse> {
         throw BadRequestException('Invalid agent ID: $agentId');
       },
     );
+    final TabledataResourceApi tabledataResourceApi = await config.createTabledataResourceApi();
 
     agent.isHealthy = isHealthy;
     agent.healthDetails = healthDetails;
     agent.healthCheckTimestamp = DateTime.now().millisecondsSinceEpoch;
 
     await datastore.db.commit(inserts: <Agent>[agent]);
+    /// Insert data to [BigQuery] whenever updating data in [Datastore] 
+    await _insertBigquery(agent, tabledataResourceApi);
 
-    return UpdateAgentHealthResponse(agent);
+    final TableDataList tableDataList = await tabledataResourceApi.list(projectId, dataset, table);
+
+    return UpdateAgentHealthResponse(agent, tableDataList.totalRows);
+  }
+
+  Future<void> _insertBigquery(Agent agent, TabledataResourceApi tabledataResourceApi) async {
+    final List<Map<String, Object>> requestRows = <Map<String, Object>>[];
+    
+    requestRows.add(<String, Object>{
+      'json': <String, Object>{
+        'Timestamp': agent.healthCheckTimestamp,
+        'AgentID': agent.agentId,
+        // TODO(keyonghan): add more detailed states https://github.com/flutter/flutter/issues/44213
+        'Status': agent.isHealthy?'Healthy':'Unhealthy',
+        'Detail': agent.healthDetails,
+      },
+    });
+
+    /// [rows] to be inserted to [BigQuery]
+    final TableDataInsertAllRequest request =
+      TableDataInsertAllRequest.fromJson(<String, Object>{
+      'rows': requestRows
+    });
+
+    try {
+      await tabledataResourceApi.insertAll(request, projectId, dataset, table);
+    } catch(ApiRequestError){
+      log.warning('Failed to add ${agent.agentId} status to BigQuery: $ApiRequestError');
+    }
   }
 }
 
 @immutable
 class UpdateAgentHealthResponse extends JsonBody {
-  const UpdateAgentHealthResponse(this.agent) : assert(agent != null);
+  const UpdateAgentHealthResponse(this.agent, this.totalRows) : assert(agent != null);
 
   final Agent agent;
+  /// [totalRows] is for test purpose only.
+  final String totalRows;
```

Comment:
By removing totalRows, a response avoids carrying an unused field in production.

Reply:
Removed.